pkg/kube/inject/inject.go

```diff
@@ -659,7 +659,8 @@ func IntoObject(sidecarTemplate Templates, valuesConfig string, revision string,
 		ObjectMeta: *metadata,
 		Spec:       *podSpec,
 	}
-	if !injectRequired(ignoredNamespaces, &Config{Policy: InjectionPolicyEnabled}, &pod.Spec, pod.ObjectMeta) {
+	filteredNamespaces := append(ignoredNamespaces, meshconfig.RootNamespace)
```

Comment:
I am not sure this preserves the behavior expected. The current behavior is that a user that has `name=.Release.Namespace` avoids injection. Note that the vast majority of users do *not* have this set, so I suspect on upgrade users would unexpectedly see the injection logic change. Additionally, .Release.Namespace != meshconfig.RootNamespace

Reply:
got it

Reply:
@howardjohn Can you elaborate more on this ? `.Release.Namespace != meshconfig.RootNamespace` 

I guess I don't what the implication of a root namespace is